pkg/trait/owner.go

```diff
@@ -78,17 +78,20 @@ func (t *ownerTrait) Apply(e *Environment) error {
 	}
 
 	e.Resources.VisitMetaObject(func(res metav1.Object) {
-		references := []metav1.OwnerReference{
-			{
-				APIVersion:         e.Integration.APIVersion,
-				Kind:               e.Integration.Kind,
-				Name:               e.Integration.Name,
-				UID:                e.Integration.UID,
-				Controller:         &controller,
-				BlockOwnerDeletion: &blockOwnerDeletion,
-			},
+		// Avoid setting owner references across namespaces (resources are asynchronously refused by the api server)
```

Comment:
I think we can even say that cross-namespace owner references are forbidden.

Reply:
Yes, I put that description because the problem is a bit more subtle than that. I don't know if it's a consequence of the server-side apply, but if you use an invalid owner, you don't get absolutely any error in the operator. The error comes later and it is only visible in the Kubernetes cluster events. If you don't watch them, you just don't find the resource that the operator thinks he has created... 

Is this expected @astefanutti ?

Reply:
hum that's a very good question :) I've never noticed it, probably because of the behaviour you describe. It seems there have been some issue with invalid references, like kubernetes/kubernetes#65200. I would not expect it to have something to do with server-side apply, but more an issue with how invalid references are handled asynchronously.

Reply:
Ok, if it's only limited to these cases of invalid references it's not a great issue after we deal with them..

Reply:
Yes I agree. The controller-runtime util package has some helpers that do some pre-flight checks client-side when updating the owner references to prevent invalid references before applying them server-side. That tends to confirm they are not reported synchronously by the API server.